src/model/APNSPayload.php

```diff
@@ -31,22 +31,41 @@ class APNSPayload {
 	/** @var array */
 	private $custom = [];
 
+	// This class can be set up too many different ways to have any kind of shared constructor, so we'll explicitly define an empty one
+	function __construct() { }
+
+	/**
+	 * Create an APNSPayload from the provided string
+	 *
+	 * @param string $string
+	 */
+	static function fromString( string $string ): APNSPayload {
+		return APNSPayload::fromAlert( APNSAlert::fromString( $string ) );
+	}
+
 	/**
-	 * Create an APNSPayload from the provided APNSAlert or string object, as well as any custom data
+	 * Create an APNSPayload from the provided APNSAlert or string object
 	 *
 	 * @param APNSAlert $alert
-	 * @param array $custom
 	 */
-	function __construct( APNSAlert $alert, array $custom = [] ) {
-		$this->alert = $alert;
-		$this->custom = $custom;
+	static function fromAlert( APNSAlert $alert ): APNSPayload {
+		$payload = new APNSPayload();
+		$payload->setAlert( $alert );
+		return $payload;
 	}
 
-	static function fromString( string $string ): APNSPayload {
-		return new APNSPayload( APNSAlert::fromString( $string ) );
+	/**
+	 * Create an APNSPayload that contains only a badge count
+	 *
+	 * @param APNSAlert $alert
+	 * @param array $custom
+	 */
+	static function fromBadgeCount( int $count ): APNSPayload {
+		return ( new APNSPayload() )
+			->setBadgeCount( $count );
```

Comment:
Nitpick. It would be great if all the `from-` factories used the same syntax, for consistency's sake, regardless of which (but this is my fave 😄)